rlclientlib/logger/http_transport_client.h

```diff
@@ -126,45 +162,55 @@ pplx::task<web::http::status_code> http_transport_client<TAuthorization>::http_r
   const auto stream = concurrency::streams::bytestream::open_istream(container);
   request.set_body(stream, container_size);
 
-  return _client->request(request).then([this, try_count](pplx::task<http_response> response) {
-    web::http::status_code code = status_codes::InternalError;
-    api_status status;
+  // lambda which examines the provided task and either 1) generates a replacement task to retry
+  // the request, or 2) passes the provided task downstream to emit its response
+  auto retry_request_on_failure_lambda = [this, try_count](
+                                             pplx::task<http_response> response_task) -> pplx::task<http_response> {
+    web::http::status_code response_code = status_codes::InternalError;
 
     try
     {
-      code = response.get().status_code();
+      response_code = response_task.get().status_code();
     }
     catch (const std::exception& e)
     {
       TRACE_ERROR(_trace, e.what());
     }
 
-    // If the response is not the expected code then it has failed. Retry if possible otherwise report background
-    // error.
-    if (code != status_codes::Created && code != status_codes::NoContent)
+    bool success = (response_code >= status_codes::OK) && (response_code < status_codes::MultipleChoices);
+    if (success) { return response_task; }
+
+    // If the response is not success class then it has failed. Retry if possible otherwise report background error.
+    auto deadline = _start_time + _max_retry_duration;
```

Comment:
would this overflow when `_max_retry_duration = std::chrono::milliseconds::max()`?